smolyay/surrogate.py

```diff
@@ -156,6 +156,118 @@ def grid_generator(self, value):
         self._grid_generator = value
         self._reset_grid()
 
+    def derivative(self, x):
```

Comment:
This should be called gradient to make it clear that it will return all components for a multivariable function.
```suggestion
    def gradient(self, x):
```